exercises/takudzwa_makoni/ex02_oo_basics/src/shapeSorter.h

```diff
@@ -0,0 +1,93 @@
+#ifndef SHAPESORTER_H
+#define SHAPESORTER_H
+#include <memory>
+#include <vector>
+
+#include "shape.h"
+
+// << OPERATOR OVERLOAD FOR SHAPE CLASS
+std::ostream& operator << (std::ostream& stream, const shape& other) {
+
+	stream << "name: " << other.name() << ", area: " << other.area() << ", perimeter: " << other.perimeter() << ", sides: " << other.numOfSides();
+	return stream;
+
+}
+
+namespace {
+
+	bool comparePerim(const std::unique_ptr<shape>& a, const std::unique_ptr<shape>& b) { return a->perimeter() < b->perimeter(); }
+	bool compareArea(const std::unique_ptr<shape>& a, const std::unique_ptr<shape>& b) { return a->area() < b->area(); }
+}
+
+
+class shapeSorter {
+
+private:
+	//int chosenNumOfSides;
+	//std::string chosenType;
+	std::vector<std::unique_ptr<shape>> shapes;
+	
+
+public:
+
+
+	void printMatchForSide(int chosenNumOfSides) {
+	
+		std::cout << "\nMATCHES FOR SIDE \"" << chosenNumOfSides <<"\"\n";
+		for (size_t i = 0; i != shapes.size(); i++)
+
+		{
+			if (shapes[i]->numOfSides() == chosenNumOfSides) { std::cout << *(shapes[i]) << " "; };
+		}
+
+		std::cout << "\n\n";
+
+	};
+
+	void printMatchForType(std::string chosenType) {
+
+		std::cout << "\nMATCHES FOR TYPE \"" << chosenType << "\"\n";
+		for (size_t i = 0; i != shapes.size(); i++){
+		std::transform(chosenType.begin(), chosenType.end(), chosenType.begin(), std::toupper);
```

Comment:
@Milli-Lab just to help you a bit with the GCC errors. change `std::toupper` to `::toupper`. The std::toupper overload does not have the signature we need.

Reply:
replaced it👍